### ISSUE TITLE
Support plan AtomicLong, IgniteSet with thin client?

### ISSUE THREAD

**Issue by pkgonan** [gridgain/gridgain#2308]
Hi. Is there any support plan AtomicLong, IgniteSet with thin client? @ptupitsyn

**Comment by ptupitsyn** [comment]
Hi. Yes, there are plans to add that.
As a workaround, you can wrap AtomicLong or IgniteSet inside a Service and
call it from the thin client.

**Comment by pkgonan** [comment]
@ptupitsyn 
Thank you for your reply. 
Is there an expected release version or date? Jira issue url are fine too.

**Comment by pkgonan** [comment]
@ptupitsyn 
```
you can wrap AtomicLong or IgniteSet inside a Service and call it from the thin client.
```
Do you have any code or sample that I can refer to?

**Comment by ptupitsyn** [comment]
@pkgonan here is an example with AtomicLong wrapped in a Service and used from thin client:

https://gist.github.com/ptupitsyn/e3e23845b0bc4de45403c7bcc7c704ae

**Comment by pkgonan** [comment]
@ptupitsyn Thank you very much :) I will try it using java language.

**Comment by ptupitsyn** [comment]
Yes, should be straightforward to translate to Java. When people ping me I assume it is about .NET :)

**Comment by pkgonan** [comment]
@ptupitsyn 
Thick Client is too heavy, so there will be many users who want to use Thin Client, but the inability to use AtomicLong and IgniteSet seems to be a big disadvantage. If users want to use AtomicLong and IgniteSet as a thin client, I think it is inconvenient to create a Jar file to wrap it with a service. The usability is not good, so I am thinking about whether I should use Redis. Will I be able to use AtomicLong and IgniteSet directly in Java Thin Client in the near future? It is too bad usability from the point of view of paid users to wrap it as a service and use it.

**Comment by ptupitsyn** [comment]
> paid users

If you are a GridGain customer, please contact the support team at https://www.gridgain.com/services/support
